src/test/java/testhelper/PerformanceUtils.java

```diff
@@ -0,0 +1,28 @@
+package testhelper;
+
+import com.sun.management.ThreadMXBean;
+
+import java.lang.management.ManagementFactory;
+
+public class PerformanceUtils {
+
+  private static ThreadMXBean THREAD_MX_BEAN;
+
+  static {
+    Object bean = ManagementFactory.getThreadMXBean();
```

Comment:
`getThreadMXBean()` returns `java.lang.management.ThreadMXBean`, but we need `com.sun.management.ThreadMXBean` to be able to call `getThreadAllocatedBytes(long)` (it's a part of a private API). This impose potential problem. This code would work only on OpenJDK/Oracle JVM. I'm not sure if this is important.